service/rpc/handler.go

```diff
@@ -0,0 +1,25 @@
+package rpc
+
+import (
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/service/header"
+	"github.com/celestiaorg/celestia-node/service/share"
+	"github.com/celestiaorg/celestia-node/service/state"
+)
+
+var log = logging.Logger("rpc")
+
+type Handler struct {
+	state  *state.Service
+	share  share.Service
```

Comment:
Shouldn't it be also a pointer?

Reply:
As you deleted `share.Service` interface, I think now it's better to pass everywhere a pointer to `share.Service` struct.

Reply:
Oh wait this will be fixed in subsequent PR, not this one

Reply:
ooops. sorry

Reply:
https://github.com/celestiaorg/celestia-node/pull/695/commits/8d729243375234f8bdbb663c7d81a18dcdaaf5e3